wasd-controls.js

```diff
@@ -18,6 +20,9 @@ export class WasdControlsComponent extends Component {
         this.down = false;
         this.left = false;
 
+        this.zero = [0, 0, 0]
+        this.direction = new Float32Array(3);
```

Comment:
You can store that "globally" in the script:

```js
const _direction = vec3.create();
```

This is safe because files are modules (isolated), and JS is monothread, so no concurrent access can take place.

Reply:
You can keep it a little more generic as well `_vector3`